storage/src/main/java/tech/pegasys/artemis/storage/rocksdb/RocksDbDatabase.java

```diff
@@ -0,0 +1,669 @@
+    try {
+      db =
+          TransactionDB.open(
+              options,
+              txOptions,
+              configuration.getDatabaseDir().toString(),
+              columnDescriptors,
+              columnHandles);
+
+      final ImmutableMap.Builder<RocksDbColumn, ColumnFamilyHandle> builder =
+          ImmutableMap.builder();
+      for (ColumnFamilyHandle columnHandle : columnHandles) {
+        final RocksDbColumn rocksDbColumn = columnsById.get(Bytes.wrap(columnHandle.getName()));
+        builder.put(rocksDbColumn, columnHandle);
+      }
+      columnHandlesByName = builder.build();
+
+    } catch (RocksDBException e) {
+      throw new DatabaseStorageException(
+          "Failed to open database at path: " + configuration.getDatabaseDir(), e);
+    }
+  }
+
+  private BlockBasedTableConfig createBlockBasedTableConfig(final RocksDbConfiguration config) {
+    final LRUCache cache = new LRUCache(config.getCacheCapacity());
+    return new BlockBasedTableConfig().setBlockCache(cache);
+  }
+
+  @Override
+  public void storeGenesis(final Store store) {
+    throwIfClosed();
+    final WriteOptions options = new WriteOptions();
+    try (Transaction transaction = db.beginTransaction(options)) {
+      try {
+        ColumnFamilyHandle defaultColumn = columnHandlesByName.get(DEFAULT);
+        transaction.put(
+            defaultColumn,
+            GENESIS_TIME_KEY.getId(),
+            Longs.toByteArray(store.getGenesisTime().longValue()));
+        transaction.put(
+            defaultColumn,
+            RocksDbDefaultColumnKey.JUSTIFIED_CHECKPOINT_KEY.getId(),
+            serialize(store.getJustifiedCheckpoint()));
+        transaction.put(
+            defaultColumn,
+            RocksDbDefaultColumnKey.BEST_JUSTIFIED_CHECKPOINT_KEY.getId(),
+            serialize(store.getBestJustifiedCheckpoint()));
+        transaction.put(
+            defaultColumn,
+            FINALIZED_CHECKPOINT_KEY.getId(),
+            serialize(store.getFinalizedCheckpoint()));
+        store
+            .getBlockRoots()
+            .forEach(
+                root -> {
+                  final SignedBeaconBlock block = store.getSignedBlock(root);
+                  final BeaconState state = store.getBlockState(root);
+                  try {
+                    addHotBlock(transaction, root, block);
+                    byte[] rootArray = root.toArrayUnsafe();
+                    transaction.put(
+                        columnHandlesByName.get(HOT_STATES_BY_ROOT), rootArray, serialize(state));
+                    transaction.put(
+                        columnHandlesByName.get(FINALIZED_ROOTS_BY_SLOT),
+                        Longs.toByteArray(block.getSlot().longValue()),
+                        rootArray);
+                    transaction.put(
+                        columnHandlesByName.get(FINALIZED_BLOCKS_BY_ROOT),
+                        rootArray,
+                        serialize(block));
```

Comment:
We're storing state and blocks in both hot and cold storage here (we should only have 1 block / state here since we're handling genesis - the `forEach` is a little misleading): see `addHotBlock` on 190 and `putFinalizedState` on line 202.  We need the latest finalized block and state in both hot and cold storage - if we don't have any hot blocks / state we won't be able to restart because the fork choice and block import logic runs on top of `RecentChainData`.

Reply:
I've added some extra comments to make this clearer.  And verified we have tests to cover this.